src/appleseed/renderer/modeling/project/projectfilereader.cpp

```diff
@@ -3082,6 +3092,68 @@ auto_release_ptr<Project> ProjectFileReader::read(
     return event_counters.has_errors() ? auto_release_ptr<Project>(0) : project;
 }
 
+auto_release_ptr<Assembly> ProjectFileReader::read_archive(
+    const char*             archive_filepath,
+    const char*             schema_filepath,
+    const SearchPaths&      search_paths,
+    const int               options)
+{
+    assert(archive_filepath);
+
+    XercesCContext xerces_context(global_logger());
+    if (!xerces_context.is_initialized())
+        return auto_release_ptr<Assembly>(0);
+
+    if ((options & OmitProjectSchemaValidation) == false && schema_filepath == 0)
+    {
+        RENDERER_LOG_ERROR(
+            "Archive schema validation enabled, but no schema filepath provided.");
+        return auto_release_ptr<Assembly>(0);
+    }
+
+    Stopwatch<DefaultWallclockTimer> stopwatch;
+    stopwatch.start();
+
+    EventCounters event_counters;
+    auto_release_ptr<Project> project(
+        load_project_file(
+            archive_filepath,
+            schema_filepath,
+            options | OmitSearchPaths,
+            event_counters,
+            &search_paths));
+
+    if (project.get())
+    {
+        if (!event_counters.has_errors() &&
+            !(options & OmitProjectFileUpdate) &&
+            project->get_format_revision() < ProjectFormatRevision)
+            upgrade_project(*project, event_counters);
+    }
+
+    stopwatch.measure();
+
+    print_loading_results(
+        archive_filepath,
+        false,
+        event_counters,
+        stopwatch.get_seconds());
+
+    if (event_counters.has_errors())
+        return auto_release_ptr<Assembly>(0);
+
+    if (project->get_scene())
+    {
+        if (Assembly* assembly = project->get_scene()->assemblies().get_by_name("assembly"))
```

Comment:
I think this is our convention. Studio, Gafferseed and appleseed-maya all name the main assembly "assembly". I'm not sure about appleseed-max.
Hardcoding the name simplifies things and I don't think it's too restrictive.
Nearly all archives will be generated by exporting from Maya / Max or Gaffer.